csrc/codegen.cpp

```diff
@@ -1469,42 +1476,34 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
       }
     }
 
-    // TODO: support ITEMS_PER_THREAD > 1
-    constexpr int items_per_thread = 1;
+    const int64_t items_per_thread = grouped_id != nullptr
+        ? grouped_id->extent()->evaluate().as<int64_t>()
+        : 1;
 
     const auto input = aop->in()->as<kir::TensorIndex>();
 
     template_args.arg(input->dtype()); // DataT
     template_args.arg(items_per_thread); // ITEMS_PER_THREAD
 
-    // Call the runtime argsort function
     ArgumentBuilder func_args;
-
-    // The output tensor is assumed to be a register tensor, and thus
-    // its storage should always be available without predication
-    NVF_ERROR_EQ(
-        output->view()->getMemoryType(),
-        MemoryType::Local,
-        "Argsort output must be a Local tensor: ",
-        output->toString());
+    // Both inputs and outputs are guaranteed to be allocated as a
+    // contiguous chunk of buffer of size items_per_thread. Predicates
+    // are not used here because out-of-bounds elements are initizlied
+    // to the max or min value accordingly.
     func_args.arg("*(int64_t(*)[")
         .append(items_per_thread)
         .append("])")
         .append("(&")
         .append(genInline(output))
         .append(")");
-
-    NVF_ERROR(aop->predicate() != nullptr && aop->predicate()->hasValue());
-    // {pred ? input : (isDescending ? min : max)}
-    func_args.arg("{")
-        .append(genInline(aop->predicate()))
-        .append(" ? ")
+    func_args.arg("*(")
```

Comment:
Since the input is initialized to max or min, it can be used as is. The code looks like below:

```
  Array<float, 4, 1> T1;
  #pragma unroll
  for(nvfuser_index_t i3 = 0; i3 < 4; ++i3) {
    T1[i3] = POS_INFINITY;
  }
  #pragma unroll
  for(nvfuser_index_t i3 = 0; i3 < 4; ++i3) {
    nvfuser_index_t i4;
    i4 = i3 + nvfuser_zero;
    if ((i2 < (-i4))) {
      T1[i3]
         = T0[(i1 + i4)];
    }
  }
  Array<int64_t, 4, 1> T2;
  argsort::blockArgsort<26, 1, 1, 0, 1, 1, float, 4>(*(int64_t(*)[4])(&T2[0]), *(float(*)[4])(&T1[0]), false, DefaultBlockDim());
```